docs/proposals/configuration.md

```diff
@@ -0,0 +1,350 @@
+
+Currently KubeEdge uses beehive's config package to analyse configuration files, but it has many drawbacks,when the program is running, it will output a lot of logs first.
+
+Is also no need to read configuration variables in multiple ways, such as commandline and env.
+
+## Goals
+
+Binary default reads yaml files in `/etc/kubeedge/config` directory and then performs the relevant logical processing.
+
+When binary run, it will first load config yaml file by `gopkg.in/yaml.v2` package rather than `beehive` package,
+
+and it also need to easily generate the configuration files for `keadm`.
+
+
+## Proposal
+
+### cloudcore 
+
+#### config struct 
+
+```go
+
+type CloudCoreConfig struct {
+	Kube             *KubeConfig             `json:"kube,omitempty"`
+	EdgeController   *EdgeControllerConfig   `json:"edgeController,omitempty"`
+	DeviceController *DeviceControllerConfig `json:"deviceController,omitempty"`
+	Cloudhub         *CloudHubConfig         `json:"cloudHub,omitempty"`
+	Modules          *Modules                `json:"modules,omitempty"`
+}
+
+type EdgeControllerConfig struct {
+	NodeUpdateFrequency int32 `json:"nodeUpdateFrequency,omitempty"`
+}
+
+type DeviceControllerConfig struct {
+}
+
+type CloudHubConfig struct {
+	// enable websocket protocol ,default true
+	EnableWebsocket bool `json:"enableWebsocket,omitempty"`
+	// open port for websocket server, default 10000
+	WebsocketPort int32 `json:"websocketPort,omitempty"`
+	// enable quic protocol, default false
+	EnableQuic bool `json:"enableQuic,omitempty"`
+	// open prot for quic server, default 10001
+	QuicPort int32 `json:"quicPort,omitempty"`
+	// the max incoming stream for quic server, default 10000
+	MaxIncomingStreams int32 `json:"maxIncomingStreams,omitempty"`
+	// enable unix domain socket protocol, default true
+	EnableUnixSocket bool `json:"enableUnixSocket,omitempty"`
+	// unix domain socket address, default unix:///var/lib/kubeedge/kubeedge.sock
+	UnixSocketAddress string `json:"unixSocketAddress,omitempty"`
+	//default 0.0.0.0
+	Address string `json:"address,omitempty"`
+	//default /etc/kubeedge/ca/rootCA.crt
+	TLSCaFile string `json:"tlsCaFile,omitempty"`
+	//default /etc/kubeedge/certs/edge.crt
+	TLSCertFile string `json:"tlsCertFile,omitempty"`
+	//default /etc/kubeedge/certs/edge.key
+	TLSPrivateKeyFile string `json:"tlsPrivateKeyFile,omitempty"`
+	//default 30
+	KeepaliveInterval int32 `json:"keepaliveInterval,omitempty"`
+	//default 30
+	WriteTimeout int32 `json:"writeTimeout,omitempty"`
+	//default 10
+	NodeLimit int32 `json:"nodeLimit,omitempty"`
+}
+
+type KubeConfig struct {
+	// kube-apiserver address (such as:http://localhost:8080)
+	Master string `json:"master,omitempty"`
+	// Enter path to kubeconfig file to enable https connection to k8s apiserver, if master and kubeconfig are both set, master will override any value in kubeconfig.
```

Comment:
We can move this comment `master will override any value in kubeconfig`  above the master field

Reply:
good